src/track-gpc-signal.ts

```diff
@@ -0,0 +1,21 @@
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { log } from '@guardian/libs';
+import { EventTimer } from './event-timer';
+
+/**
+ * Collect metrics on gpcSignal presence and value
+ * https://globalprivacycontrol.github.io/gpc-spec/
+ */
+const initTrackGpcSignal = (consentState: ConsentState) => {
+	// If undefined we set the property value to -1, false is 0, true is 1
+	const gpcSignal =
+		consentState.gpcSignal === undefined ? -1 : +consentState.gpcSignal;
```

Comment:
Minor point for a follow up but if we always expect the `gpcSignal` property to be present should the type from CMP be `gpcSignal: undefined | boolean` rather than optional?

Reply:
I agree, it should be `gpcSignal: undefined | boolean` as you describe. Are you happy to merge this and i'll follow up with a fix on the CMP lib afterwards?

Reply:
Sure that's fine